include/nighthawk/user_defined_output/user_defined_output_plugin.h

```diff
@@ -0,0 +1,107 @@
+#pragma once
+
+#include <memory>
+
+#include "envoy/buffer/buffer.h"
+#include "envoy/common/pure.h"
+#include "envoy/config/typed_config.h"
+#include "external/envoy/source/common/common/statusor.h"
+
+#include "external/envoy/source/common/http/header_map_impl.h"
+
+
+namespace Nighthawk {
+
+// Information about a Nighthawk worker thread. May expand to contain more fields over time as
+// desired.
+struct WorkerMetadata {
+  std::string name;
+};
+
+/**
+ * An interface for the UserDefinedOutputPlugin that receives responses and allows users to
+ * attach their own custome output to each worker Result.
+ *
+ */
+class UserDefinedOutputPlugin {
```

Comment:
Where are we documenting the thread-safety requirements for plugin implementations?